### PR TITLE
links: add missing check

### PR DESCRIPTION
**Pull request by curioussavage**
the else case in the formatBody method was not checking
if some properties existed before accessing

### CODE REVIEW

**Comment by curioussavage** [comment]
I think there is another bug that affected this. here is a stack trace from my dev instance.

```SERVER-Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.103 Safari/537.36	props.listings.map is not a function	/	    at ListingList.render (/home/vagrant/mobile/app/src/views/components/ListingList.jsx:143:22)
Warning: Failed propType: Required prop `status` was not specified in `ErrorPage`.
Warning: Failed propType: Required prop `originalUrl` was not specified in `ErrorPage`.
Caught exception [TypeError: reject is not a function] TypeError: reject is not a function
    at /home/vagrant/mobile/app/node_modules/superagent/lib/node/index.js:1036:11
    at Request.callback (/home/vagrant/mobile/app/node_modules/superagent/lib/node/index.js:781:12)
    at ClientRequest.<anonymous> (/home/vagrant/mobile/app/node_modules/superagent/lib/node/index.js:736:10)
    at emitOne (events.js:77:13)
    at ClientRequest.emit (events.js:169:7)
    at Socket.socketErrorListener (_http_client.js:259:9)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at emitErrorNT (net.js:1253:8)
    at doNTCallback2 (node.js:439:9)
SERVER	reject is not a function	NO REQUEST URL	    at /home/vagrant/mobile/app/node_modules/superagent/lib/node/index.js
Caught exception { [AssertionError: Resource leak detected.]
  name: 'AssertionError',
  actual: false,
  expected: true,
  operator: '==',
  message: 'Resource leak detected.',
  generatedMessage: false } AssertionError: Resource leak detected.
    at removeWorker (cluster.js:328:9)
    at ChildProcess.<anonymous> (cluster.js:348:34)
    at ChildProcess.g (events.js:260:16)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
SERVER	Resource leak detected.	NO REQUEST URL	    at removeWorker (cluster.js```

**Comment by curioussavage** [comment]
:eyeglasses: @ajacksified

File: src/endpoints/v1.es6.js [hunk 1]
```diff
@@ -829,13 +829,13 @@ class APIv1Endpoint {
...
         return function(body) {
-          if (body.data && body.data.children && body.data.children[0]) {
+          if (has(body, 'data.children.0')) {
             if (options.id) {
               return new Link(body.data.children[0].data).toJSON();
             } else {
               return body.data.children.map(c => new Link(c.data).toJSON());
             }
-          } else if (body.data.children) {
+          } else if (has(body, 'data.children')) {
```
Comment (curioussavage) [inline src/endpoints/v1.es6.js]: looking at this again I should probably get rid of the else and just check for data.children above.

**Comment by ajacksified** [comment]
screen shot 2017-05-17 at 12 11 34 pm

adam stahp

**Comment by curioussavage** [comment]
lol. sorry.

Just trying to clear harold

**Pull request closed without merging**